numerical_methods/matrix_transpose.c

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file
+ * @author [jerr-it](https://github.com/jerr-it)
+ * @brief This is a C implementation of the [transpose of a matrix](https://en.wikipedia.org/wiki/Transpose)
+ * @details In linear algebra, the transpose of a matrix is an operator which flips a matrix over its diagonal; 
+ * that is, it switches the row and column indices of the matrix A by producing another matrix, often denoted by A^T
+ */
+
+#include <assert.h>  /// for assert
+#include <stdbool.h> /// for bool
+#include <stddef.h>  /// for size_t
+#include <stdlib.h>  /// for malloc, free
+
+/**
+ * @brief Matrix struct for dynamically allocated memory
+ */
+typedef struct {
+    int rows;
+    int cols;
+    double *data;
```

Comment:
Please add a one-line description of what the variable is for (**see the example below**).

```c
int64_t data = 0;       ///< The value/key of the node.
```